aspnetcore/fundamentals/configuration/index.md

```diff
@@ -1,452 +1,1504 @@
+
+The following example extracts the string value from configuration with the key `NumberKey`, types the value as an `int`, and stores the value in the variable `intValue`. If `NumberKey` isn't found in the configuration keys, `intValue` receives the default value of `99`:
+
+```csharp
+var intValue = config.GetValue<int>("NumberKey", 99);
 ```
 
-Note: If `-key1` isn't present in the [switch mappings](#switch-mappings) given to the configuration provider, a `FormatException` is thrown.
+## GetSection, GetChildren, and Exists
 
-### Duplicate keys
+For the examples that follow, consider the following JSON file. Four keys are found across two sections, one of which includes a pair of subsections:
 
-If duplicate keys are provided, the last key-value pair is used.
+```json
+{
+  "section0": {
+    "key0": "value",
+    "key1": "value"
+  },
+  "section1": {
+    "key0": "value",
+    "key1": "value"
+  },
+  "section2": {
+    "subsection0" : {
+      "key0": "value",
+      "key1": "value"
+    },
+    "subsection1" : {
+      "key0": "value",
+      "key1": "value"
+    }
+  }
+}
+```
 
-### Switch mappings
+When the file is read into configuration, the following unique hierarchical keys are created to hold the configuration values:
 
-When manually building configuration with `ConfigurationBuilder`, a switch mappings dictionary can be added to the `AddCommandLine` method. Switch mappings allow key name replacement logic.
+* section0:key0
+* section0:key1
+* section1:key0
+* section1:key1
+* section2:subsection0:key0
+* section2:subsection0:key1
+* section2:subsection1:key0
+* section2:subsection1:key1
 
-When the switch mappings dictionary is used, the dictionary is checked for a key that matches the key provided by a command-line argument. If the command-line key is found in the dictionary, the dictionary value (the key replacement) is passed back to set the configuration. A switch mapping is required for any command-line key prefixed with a single dash (`-`).
+### GetSection
 
-Switch mappings dictionary key rules:
+[IConfiguration.GetSection](/dotnet/api/microsoft.extensions.configuration.iconfiguration.getsection) extracts a configuration subsection with the specified subsection key.
 
-* Switches must start with a dash (`-`) or double-dash (`--`).
-* The switch mappings dictionary must not contain duplicate keys.
+To return an [IConfigurationSection](/dotnet/api/microsoft.extensions.configuration.iconfigurationsection) containing only the key-value pairs in `section1`, call `GetSection` and supply the section name:
+
+```csharp
+var configSection = _config.GetSection("section1");
+```
+
+Similarly, to obtain the values for keys in `section2:subsection0`, call `GetSection` and supply the section path:
+
+```csharp
+var configSection = _
```

Comment:
Specify where in the `Startup` class. I assume its the constructor.

Reply:
> Specify where in the Startup class. I assume its the constructor.

Something has gone peculiar with the lines where your remarks are attached ... here and back on your `Get<T>` vs. `Bind` comment.